pkg/controller/noderef/noderef_controller.go

```diff
@@ -165,8 +169,8 @@ func (r *ReconcileNodeRef) reconcile(ctx context.Context, cluster *v1alpha2.Clus
 	nodeRef, err := r.getNodeReference(corev1Client, providerID)
 	if err != nil {
 		if err == ErrNodeNotFound {
-			klog.Warningf("Cannot find a matching Node for Machine %q in namespace %q, retrying later", machine.Name, machine.Namespace)
-			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			klog.Warningf("Failed to assign NodeRef to Machine %q: cannot find a matching Node in namespace %q, retrying later", machine.Name, machine.Namespace)
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, err
```

Comment:
I think this block would be cleaner like this:
```go
nodeRef, err := r.getNodeReference(corev1Client, providerID)
if err != nil {
  return reconcile.Result{}, err
}
```

And then above, in `Reconcile()`, do this:
```go
result, err := r.reconcile(ctx, cluster, machine)
if err != nil {
  if err == ErrNodeNotFound {
    klog.Warningf("Failed to assign NodeRef to Machine %q: cannot find a matching Node in namespace %q, retrying later", machine.Name, machine.Namespace)
    return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
  }
}

Thoughts?

Reply:
I would add a `return reconcile.Result{}, err` after the `if err == ErrNodeNotFound` block, but yes, this is more clear.

Reply:
Right, thanks, will edit!

Reply:
I tried to keep code as much as possible the same. I changed the logging so that you didn't end up with a warning followed by an error, as discussed with @chuckha over slack. I will do as requested later today.

Reply:
@ncdc how about
```golang
result, err := r.reconcile(ctx, cluster, machine)
if err != nil {
	if err == ErrNodeNotFound {
		klog.Warningf("Failed to assign NodeRef to Machine %q: cannot find a matching Node in namespace %q, retrying later", machine.Name, machine.Namespace)
		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
	}

	klog.Errorf("Failed to assign NodeRef to Machine %q: %v", request.NamespacedName, err)
	r.recorder.Event(machine, apicorev1.EventTypeWarning, "FailedSetNodeRef", err.Error())
	return result, err
}
```
?

Reply:
@pires I'd change `return result, err` to `return reconcile.Result{}, err`

Reply:
Well, `result` is already `reconcile.Result{}` as per the changes to `reconcile()` but I guess you and @chuckha want to be really explicit here. By the way, when `Reconcile()` returns an `err` that is not `nil` IIRC there will be no use for the `result` so it could well be `nil`.

Reply:
Done.

Reply:
It can't be `nil` because the `result` return value isn't a pointer...

Reply:
Ah, right! Thanks.